streams/src/test/java/org/apache/kafka/streams/processor/internals/StandbyTaskTest.java

```diff
@@ -167,8 +167,7 @@ public void testStorePartitions() throws IOException {
         StreamsConfig config = createConfig(baseDir);
         StandbyTask task = new StandbyTask(taskId, topicPartitions, topology, consumer, changelogReader, config, null, stateDirectory);
         task.initialize();
-        assertEquals(Utils.mkSet(partition2), new HashSet<>(task.checkpointedOffsets().keySet()));
-
+        assertEquals(Utils.mkSet(partition2, partition1), new HashSet<>(task.checkpointedOffsets().keySet()));
```

Comment:
Hmmm... This make we wonder if the PR is correct. For in-memory stores, I think we should not write a checkpoint file -- assume an in-memory `StandbyTask` crashed and later recovers on the same host. For this case, we need to make sure that we read the whole changelog topic. A checkpoint would indicate that a part of the changelog is already restored what is not the case there.

Reply:
Also, can we rename the test?

Reply:
This is actually correct. The checkpoint file is written for all stores in the task, however, if the store is not persistent it won't have a checkpoint and the value will be -1. 
However on looking into the whole restoration of standby tasks i think it is completely broken! i.e, we register a `StateRestoreCallback` for a topic not a partition. How does that work?

Reply:
Ah. Important detail. Thanks for clarification. Than we should be good.